File: code/back/src/main/java/com/biblioo/community/domain/model/BookVotingOption.java
package com.biblioo.community.domain.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(
    name = "community_voting_options",
    indexes = {@Index(name = "idx_cvo_voting", columnList = "voting_id")})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BookVotingOption {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "voting_id", nullable = false)
  private Long votingId;

  @Column(name = "book_id", nullable = false)
  private Long bookId;

  @Column(name = "book_title", nullable = false, length = 500)
  private String bookTitle;

  @Column(name = "book_cover_url", length = 1000)
  private String bookCoverUrl;

  @Column(name = "vote_count", nullable = false)
  @Builder.Default
  private Integer voteCount = 0;
}

File: code/mobile/lib/features/community/domain/book_voting_option.dart
class BookVotingOption {
  final int id;
  final int votingId;
  final int bookId;
  final String bookTitle;
  final String? bookCoverUrl;
  final int voteCount;
  final double percentage;

  const BookVotingOption({
    required this.id,
    required this.votingId,
    required this.bookId,
    required this.bookTitle,
    this.bookCoverUrl,
    required this.voteCount,
    required this.percentage,
  });

  BookVotingOption copyWith({
    int? id,
    int? votingId,
    int? bookId,
    String? bookTitle,
    String? bookCoverUrl,
    int? voteCount,
    double? percentage,
  }) {
    return BookVotingOption(
      id: id ?? this.id,
      votingId: votingId ?? this.votingId,
      bookId: bookId ?? this.bookId,
      bookTitle: bookTitle ?? this.bookTitle,
      bookCoverUrl: bookCoverUrl ?? this.bookCoverUrl,
      voteCount: voteCount ?? this.voteCount,
      percentage: percentage ?? this.percentage,
    );
  }
}
